src/GovUk.Education.ExploreEducationStatistics.Data.Processor/Services/ImporterService.cs

```diff
@@ -181,14 +181,27 @@ await _dataImportService.UpdateStatus(dataImport.Id,
                             .Select(f => f.AutoSelectFilterItemLabel)
                             .SingleOrDefault() ?? "Total"; // If meta file didn't specify a default, look for "Total"
 
-                        return filterItems
+                        var filterItemsWithAutoSelectLabel = filterItems
                             .Where(item =>
-                                item.FilterGroup.FilterId == filterId
-                                && item.Label.Equals(autoSelectFilterItemLabel, OrdinalIgnoreCase))
-                            .Select(item => new { item.Id , item.Label})
-                            // There might be two filter items with the same label under different groups.
-                            // If so, we set no autoSelectFilterItem.
-                            .SingleOrDefault();
+                                item.FilterGroup.FilterId == filterId &&
+                                item.Label.Equals(autoSelectFilterItemLabel, OrdinalIgnoreCase));
+
+                        // There might be two or more filter items with the same label under different groups
+                        if (filterItemsWithAutoSelectLabel.Count() > 1)
+                        {
+                            // If so, does one belong to a filter group with the same label?
+                            var filterItemWithMatchingGroupLabel = filterItemsWithAutoSelectLabel.SingleOrDefault(fi => fi.FilterGroup.Label == autoSelectFilterItemLabel);
```

Comment:
Not sure this is entirely correct. Think we're missing the following two cases with this logic:

1. The filter may only have one filter group (typically a 'Default' group but could be something else). If there is only a single filter group then we should be able to find the auto-select filter item as there's no ambiguity.

2. As we can now configure the auto-select via the `filter_default` column, the auto-select labels for the filter item and filter group may differ. For example, the filter item label could be 'All', but the filter group label might be 'Total'.

   For now, given we're preventing the filter hierarchies and filter groups being used together, we could just search for a filter group with a 'Total` label.